### PR TITLE
Updates openlayers to v7 and geostyler-style to v7

### PR DESCRIPTION
**Pull request by KaiVolland**
## Description

This updates `openlayers` to [v7](https://github.com/openlayers/openlayers/releases/tag/v7.0.0).
It also updates `geostyler-style` to [v7](https://github.com/geostyler/geostyler-style/releases/tag/v7.0.0) and introduces parsing/evaluating of the new (Function)Expressions.

## Related issues or pull requests

Please list issues or pull requests that the changes you propose are related to. It does not matter if they are still open and/or unmerged, any link is appreciated.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [BSD 2-Clause License](https://github.com/geostyler/geostyler/blob/master/)
- [x] I have followed the [guidelines for contributing](https://github.com/geostyler/geostyler/blob/master/CONTRIBUTING.md)
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/geostyler/geostyler/blob/master/CODE_OF_CONDUCT.md)
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: src/Util/OlStyleUtil.ts [hunk 1]
```diff
@@ -273,6 +297,208 @@ class OlStyleUtil {
...
+      case 'strEndsWith':
+        return (args[0] as string).endsWith(args[1] as string);
+      case 'strEqualsIgnoreCase':
+        return (args[0] as string).toLowerCase() === (args[1] as string).toLowerCase() ;
+      case 'strMatches':
+        return new RegExp(args[1] as string).test(args[0] as string);
+      case 'strStartsWith':
+        return (args[0] as string).startsWith(args[1] as string);
+      default:
+        return true;
```
Comment (jansule) [inline src/Util/OlStyleUtil.ts:354]: What's the reasoning to return `true` instead of `false` by default?

**Review by hwbllmnn — APPROVED** [review]

File: src/Util/OlStyleUtil.ts [hunk 2]
```diff
@@ -273,6 +297,208 @@ class OlStyleUtil {
...
+        // https://stackoverflow.com/a/32589289/10342669
+        var splitStr = (args[0] as string).toLowerCase().split(' ');
+        for (let part of splitStr) {
+          part = part.charAt(0).toUpperCase() + part.substring(1);
+        }
+        return splitStr.join(' ');
+      case 'strConcat':
+        return args.join();
+      case 'strDefaultIfBlank':
+        return (args[0] as string)?.length < 1 ? args[1] as string : args[0] as string;
```
Comment (jansule) [inline src/Util/OlStyleUtil.ts:466]: Are you saving characters here? :grinning:

**Review by jansule — COMMENTED** [review]
Very nice @KaiVolland Just a few questions

File: src/Util/OlStyleUtil.ts [hunk 3]
```diff
@@ -273,6 +297,208 @@ class OlStyleUtil {
...
+      if (isGeoStylerFunction(arg)) {
+        return OlStyleUtil.evaluateFunction(arg, feature);
+      }
+      return arg;
+    });
+    switch (func.name) {
+      case 'between':
+        return (args[0] as number) >= (args[1] as number) && (args[0] as number) <= (args[2] as number);
+      case 'double2bool':
+        return args[0] === 0;
```
[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/Util/OlStyleUtil.ts:340]: I just implemented it as the GeoServer docs says. I don't get it at all and maybe we can remove it from the style.

File: src/Util/OlStyleUtil.ts (hunk 1 above)
Comment (KaiVolland) [reply src/Util/OlStyleUtil.ts:354]: No reason. Would you prefer `false`?

File: src/Util/OlStyleUtil.ts (hunk 2 above)
Comment (jansule) [reply src/Util/OlStyleUtil.ts:466]: `length === 0` vs `length < 1`

File: src/Util/OlStyleUtil.ts [hunk 4]
```diff
@@ -273,6 +297,208 @@ class OlStyleUtil {
...
+  public static evaluateBooleanFunction(func: GeoStylerBooleanFunction, feature?: OlFeature): boolean {
+    const args = func.args.map(arg => {
+      if (isGeoStylerFunction(arg)) {
+        return OlStyleUtil.evaluateFunction(arg, feature);
+      }
+      return arg;
+    });
+    switch (func.name) {
+      case 'between':
+        return (args[0] as number) >= (args[1] as number) && (args[0] as number) <= (args[2] as number);
```
[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/Util/OlStyleUtil.ts:338]: Yes: https://docs.geoserver.org/stable/en/user/filter/function_reference.html#comparison-functions

File: src/Util/OlStyleUtil.ts [hunk 5]
```diff
@@ -273,6 +297,208 @@ class OlStyleUtil {
 
     return template;
   }
+
+  public static evaluateFunction(func: GeoStylerFunction, feature?: OlFeature): PropertyType {
+    if (func.name === 'property') {
+      if (!feature) {
+        // TODO: throw errror instead
```
[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/Util/OlStyleUtil.ts]: 790aa0b

File: src/Util/OlStyleUtil.ts (hunk 1 above)
Comment (jansule) [reply src/Util/OlStyleUtil.ts:353]: To me, it feels a little more familiar. But I do not have a strong opinion on this. So it is ok for me, if we keep it as is

File: src/Util/OlStyleUtil.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (jansule) [reply src/Util/OlStyleUtil.ts:339]: Maybe there is a typo in the docs. I do not get the reasoning behind the current implementation. Maybe we can double check with a working example in geoserver?

[the remark this answers is not in the document]
Comment (KaiVolland) [reply src/Util/OlStyleUtil.ts]: adfecb1

File: src/Util/OlStyleUtil.ts (hunk 1 above)
Comment (KaiVolland) [reply src/Util/OlStyleUtil.ts]: adfecb1

**Review by jansule — APPROVED** [review]
Thanks for addressing the remarks

**Pull request merged**